File: myFilesForDebuggingAndInformations/analyze_sniffer.dart
import 'dart:io';
import 'dart:convert';

void main() async {
  var file = File(
      r"c:\FlutterProjects\Ringularity\Colmi_Rxx_flutter_companion\btSnifferLogsAndTimestamps\btSnifferSyncProcessAsText3.txt");
  if (!await file.exists()) {
    print("File not found: ${file.path}");
    return;
  }

  print("Analyzing ${file.path}...");
  var lines = await file.readAsLines();

  var pattern = RegExp(r"Value:\s*([0-9a-fA-F]+)");

  List<String> foundBc = [];
  List<String> found7a = [];
  List<String> foundAny = [];

  for (int i = 0; i < lines.length; i++) {
    var match = pattern.firstMatch(lines[i]);
    if (match != null) {
      String hexVal = match.group(1)!.toLowerCase();
      String entry = "Line ${i + 1}: $hexVal";
      foundAny.add(entry);

      if (hexVal.startsWith('bc27') ||
          (hexVal.startsWith('bc') && hexVal.length > 2)) {
        foundBc.add(entry);
      } else if (hexVal.startsWith('7a')) {
        found7a.add(entry);
      }
    }
  }

  print("Total 'Value:' lines found: ${foundAny.length}");

  print("\n--- BC Matches (BigData) ---");
  for (var line in foundBc.take(20)) {
    print(line);
  }

  print("\n--- 7A Matches (Legacy Sleep) ---");
  for (var line in found7a.take(20)) {
    print(line);
  }

  if (foundBc.isEmpty && found7a.isEmpty) {
    print("\n--- Sample of other Values ---");
    for (var line in foundAny.take(20)) {
      print(line);
    }
  }
}

File: myFilesForDebuggingAndInformations/analyze_sniffer.py

import re

file_path = r"c:\FlutterProjects\Ringularity\Colmi_Rxx_flutter_companion\btSnifferLogsAndTimestamps\btSnifferSyncProcessAsText3.txt"

def analyze():
    print(f"Analyzing {file_path}...")
    try:
        with open(file_path, 'r', encoding='utf-8', errors='ignore') as f:
            lines = f.readlines()
    except Exception as e:
        print(f"Error reading file: {e}")
        return

    pattern = re.compile(r"Value:\s*([0-9a-fA-F]+)")
    
    found_bc = []
    found_7a = []
    found_any = []

    for i, line in enumerate(lines):
        match = pattern.search(line)
        if match:
            hex_val = match.group(1).lower()
            found_any.append((i+1, hex_val))
            
            if hex_val.startswith('bc27') or hex_val.startswith('bc'):
                found_bc.append((i+1, hex_val))
            elif hex_val.startswith('7a'):
                found_7a.append((i+1, hex_val))

    print(f"Total 'Value:' lines found: {len(found_any)}")
    
    print("\n--- BC Matches (BigData) ---")
    for ln, val in found_bc[:20]:
         print(f"Line {ln}: {val}")
    
    print("\n--- 7A Matches (Legacy Sleep) ---")
    for ln, val in found_7a[:20]:
         print(f"Line {ln}: {val}")

    if not found_bc and not found_7a:
        print("\n--- Sample of other Values ---")
        for ln, val in found_any[:20]:
             print(f"Line {ln}: {val}")

if __name__ == "__main__":
    analyze()
